JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,598 @@
+# React Guidelines and Best Practices
+
+*Guides are not rules and should not be followed blindly. Use your head and think.*
+
+### File Organization and Naming
+As it is a good practice to divide React components as container and presentational components, you should place them in separate folders (this example uses **containers** and **components** for folder naming, but there are other naming conventions, depending on the use case and preference of developers).
+
+```
+...
+├─ components
+├─ containers 
+...
+```
+
+Use **camelCase** for general folder naming and **PascalCase** for React components, their stylesheet files and their containing folders.
+
+```
+...
+├─ components
+⎮   ├─ LoginContent
+⎮   ⎮   └─ LoginContent.js
+⎮   ⎮   └─ LoginContent.scss
+⎮   └─ LoginHeader
+⎮       └─ LoginHeader.js
+⎮       └─ LoginHeader.scss
+├─ containers 
+⎮   └─ Login
+⎮       └─ Login.js
+⎮       └─ Login.scss 
+...
+```
+
+### Component Declaration
+
+For declaring class components:
+- use ES2015 **class syntax**
+- don't use *React.createClass*
+
+``` jsx
+// bad
+export const ComponentName = React.createClass({
+  // ...
+});
+
+// good
+export class ComponentName extends React.Component {
+  // ...
+}
+```
+
+If components are stateless (they don't change state), use **functional components**:
+
+``` jsx
+export function FunctionalComponent({message}) {
+  // ...
+};
+
+// or
+
+export const FunctionalComponent = ({message}) => (
+  // ...
+);
+```
+
+### Formatting
+
+Use self-closing tags, separated with an empty space, for elements that don't have children.
+
+``` jsx
+// bad
+<ChildComponent onClick={this.onClickHandler}></ChildComponent>
+
+// good
+<ChildComponent onClick={this.onClickHandler} />
+```
+
+If there is more than one prop, separate them with line breaks and also, put opening and closing tags in separate lines.
```

Comment:
Grammar fix: If there is more than one prop, separate them with line breaks, and also put opening and closing tags on separate lines.